### PR TITLE
SEAB-5103: Notebook support in RSS Feed

### PR DESCRIPTION
**Pull request by hyunnaye**
**Description**
This PR adds notebooks to the RSS feed. It also tests that the sitemap also includes notebooks correctly.

**Review Instructions**
Verify the new integration test pass and notebooks are included in RSS feed.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-5164

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/NotebookIT.java [hunk 1]
```diff
@@ -214,7 +215,30 @@ void testStarringNotebook() {
         notebook = workflowsApi.getWorkflow(notebookID, "");
         assertEquals(0, notebook.getStarredUsers().size());
     }
+    @Test
+    void testNotebookRSSFeedAndSitemap() {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false, testingPostgres);
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/NotebookIT.java:220]: This pre-dates this PR, but this class inherits from `BaseIT` which has 
`public void resetDBBetweenTests()` which resets the DB with simple test data. 
Then if I'm not mistaken, every test in here uses `CommonTestUtilities.cleanStatePrivate2(SUPPORT, false, testingPostgres);` which may mean that the DB is being reset twice for each test. 

Maybe you should just override `resetDBBetweenTests` to reset with `cleanStatePrivate2`

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/NotebookIT.java [hunk 2]
```diff
@@ -214,7 +215,30 @@ void testStarringNotebook() {
...
+        CreateContent createContent = new CreateContent().invoke();
+        long notebookID = createContent.getNotebookID();
+
+        // There should be 1 notebook
+        assertEquals(1, notebookDAO.findAllPublishedPaths().size());
+        assertEquals(1, notebookDAO.findAllPublishedPathsOrderByDbupdatedate().size());
+
+        final MetadataApi metadataApi = new MetadataApi(openApiClient);
+        String rssFeed = metadataApi.rssFeed();
+        System.out.println(rssFeed);
```
Comment (coverbeck) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/NotebookIT.java:236]: Looks like a debugging statement? It doesn't hurt, but not really adding anything.

**Review by coverbeck — APPROVED** [review]
Pending resolution of Denis' comment.

**Review by svonworl — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java [hunk 3]
```diff
@@ -262,10 +264,12 @@ private String getRSS() {
                 Collectors.toList());
         List<AppTool> appTools = appToolDAO.findAllPublishedPathsOrderByDbupdatedate().stream().map(RSSAppToolPath::getAppTool).collect(
                 Collectors.toList());
+        List<Notebook> notebooks = notebookDAO.findAllPublishedPathsOrderByDbupdatedate().stream().map(RSSNotebookPath::getNotebook).collect(Collectors.toList());
```
Comment (kathy-t) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java:267]: Code smell: can use `toList()` instead (same for lines 263 - 266)

**Review by kathy-t — APPROVED** [review]

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**